### DISCUSSION TITLE
Cursor Timeout

### DISCUSSION THREAD

**Discussion by thairanaru** [YaLTeR/niri#284] [Ideas]
It would be cool if there was an add to make the cursor hide after a certain time period when not moving. I remember Hyprland has something like it.

**Comment by YaLTeR** [comment]
I guess that's not too difficult to implement; why though? Usually applications handle it themselves when needed.

**Reply by thairanaru** [reply]
Well, I was thinking about neovim mainly (kitty as terminal). However, sometimes it is nice to not have the cursor to distract while reading or something in a web browser for example.

**Reply by Schweber** [reply]
I also love this feature in hyprland because i mostly use the keyboard, also in the browser with vimium. With this option, the cursor is out of the way and doesn't bother me any more.

**Reply by cyuria** [reply]
I'm possibly a bit late here, but I fully agree with this. Not all applications will handle this, and my workflow is very much focused on using one of the keyboard or the trackpad/mouse but never both, which makes it an annoying distraction. IMO this wouldn't be too much of an issue, except that the cursor reappears when I switch windows, and I find that fairly distracting.

**Reply by YaLTeR** [reply]
Nah you're not late, it's good to have more use case examples on feature requests like this where it's not obvious why it is needed

**Reply by aberter0x3f** [reply]
Fully support this idea. In editors like VSCode where cursor auto-hiding is not supported, having an option to hide the cursor after a period of inactivity can significantly reduce visual distractions, enhancing focus during tasks such as reading or writing.

Additionally, it would be beneficial if the cursor could also hide while typing, further minimizing disruptions and improving user experience. This dual approach would cater to both scenarios where the cursor is not needed for extended periods and when focused keyboard input is required.

**Reply by aberter0x3f** [reply]
A pull request for the hide-on-key-press solution: https://github.com/YaLTeR/niri/pull/709

**Reply by Schweber** [reply]
Thank you @yzy-1 🙂. It would be very kind of you if you would also implement the auto-hiding after a timeout 💪, YaLTeR pointed out a possible solution in your PR.

**Reply by aberter0x3f** [reply]
Implemented auto-hiding after a timeout in PR #709.

**Reply by Schweber** [reply]
Thanks!

**Comment by YaLTeR** [comment]
It is now merged.
